src/mcts/search.cc

```diff
@@ -106,26 +106,36 @@ class MEvaluator {
     }
   }
 
-  float GetM(const EdgeAndNode& child, float q) const {
+  float GetMUtility(const EdgeAndNode& child, float q) const {
     if (!enabled_ || !parent_within_threshold_) return 0.0f;
     const float child_m = child.GetM(parent_m_);
     float m = std::clamp(m_slope_ * (child_m - parent_m_), -m_cap_, m_cap_);
     m *= FastSign(-q);
+    if (q_threshold_ > 0.0f && q_threshold_ < 1.0f) {
+      // This allows a smooth M effect with higher q thresholds, which is
+      // necessary for using MLH together with contempt.
+      q = std::max(0.0f, (std::abs(q) - q_threshold_)) / (1.0f - q_threshold_);
+    }
     m *= a_constant_ + a_linear_ * std::abs(q) + a_square_ * q * q;
     return m;
   }
 
-  float GetM(Node* child, float q) const {
+  float GetMUtility(Node* child, float q) const {
```

Comment:
I don't really like that this code is copypasted; probably we should make this one the actual one, and turn the other into a wrapper function.

Reply:
There are only two calls of the other version and one of this. Just keeping this variation and adjusting the callers seems cleaner.

Reply:
Sounds like the better choice for sure. Can you check whether it makes a difference for the DAG compatibility which of the two we keep?

Reply:
The exact same patch applies to both master and dag-update and seems to work fine.